### DISCUSSION TITLE
wrapping the layout.js in next 14 project using zustand createContext. Still I am not able to use store and its state in server side component of next js 14

### DISCUSSION THREAD

**Discussion by sandeepbfnp** [pmndrs/zustand#2313] [Bug report]
wrapping the layout.js in next 14 project using zustand createContext. Still I am not able to use store and its state in server side component of next js 14.
/app/layout.jsx
import type { Metadata } from "next";
import { Inter } from "next/font/google";
import "./globals.css";
import Header from "@/components/header";
import ZustandProvider from "../hydrationZustand";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Create Next App",
  description: "Generated by create next app",
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <ZustandProvider>
      <html lang="en">
        <body className={inter.className}>
          <Header />
          {children}
        </body>
      </html>
    </ZustandProvider>
  );
}
after this i am not able to use the store in my /app/page.jsx
giving error of using useStore in server components

**Comment by dai-shi** [comment]
You can't use it the same as you can't use `useState`.
